lib/linter.js

```diff
@@ -736,10 +769,16 @@ const BASE_TRAVERSAL_CONTEXT = Object.freeze(
  * @param {string} filename The reported filename of the code
  * @returns {Problem[]} An array of reported problems
  */
-function runRules(sourceCode, configuredRules, ruleMapper, parserOptions, parserName, settings, filename) {
+function runRules(traversalParameters) {
     const emitter = createEmitter();
     const traverser = new Traverser();
 
+    const sourceCode = traversalParameters.sourceCode;
+    const parserOptions = traversalParameters.parserOptions;
+    const parserName = traversalParameters.parserName;
+    const settings = traversalParameters.settings;
+    const filename = traversalParameters.filename;
```

Comment:
I eagerly await being able to can use destructuring in this codebase when we drop Node 4 support 😄